docs/advanced/additional-requests.rst

```diff
@@ -0,0 +1,432 @@
+.. _`advanced-requests`:
+
+===================
+Additional Requests
+===================
+
+Websites nowadays needs a lot of page interactions to display or load some key
+information. In most cases, these are done via AJAX requests. Some examples of these are:
+
+    * Clicking a button on a page to reveal other similar products.
+    * Clicking the `"Load More"` button to retrieve more images of a given item.
+    * Scrolling to the bottom of the page to load more items `(i.e. infinite scrolling)`.
+    * Hovering on a certain webpage element that reveals a tool-tip containing
+      additional page info.
+
+As such, performing additional requests inside Page Objects are inevitable to
+properly extract data for some websites.
+
+.. warning::
+
+    Additional requests made inside a Page Object aren't meant to represent
+    the **Crawling Logic** at all. They are simply a low-level way to interact
+    with today's websites which relies on a lot of page interactions to display
+    its contents.
+
+
+HttpRequest
+===========
+
+Additional requests are defined using a simple data container that represents
+a generic HTTP Request: :class:`~.HttpRequest`. Here's an example:
+
+.. code-block:: python
+
+    import json
+    import web_poet
+
+    request = web_poet.HttpRequest(
+        url="https://www.api.example.com/product-pagination/",
+        method="POST",
+        headers={
+            "Host": "www.example.com",
+            "Content-Type": "application/json; charset=UTF-8",
```

Comment:
perhaps something like a `json` keyword in `post()` could set this up automatically similar to the requests lib.